### PR TITLE
fix: prefetch flag implementation

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
refactor and tests

**Did you add tests for your changes?**
yes
**If relevant, did you update the documentation?**

**Summary**
refactor AdvancedGroup, --hot and --prefetch flag implementation and corresponding tests

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: packages/webpack-cli/lib/groups/AdvancedGroup.js [hunk 1]
```diff
@@ -13,21 +13,12 @@ class AdvancedGroup extends GroupHelper {
         if (args.hot) {
             const { HotModuleReplacementPlugin } = require('webpack');
             const hotModuleVal = new HotModuleReplacementPlugin();
-            if (options && options.plugins) {
-                options.plugins.unshift(hotModuleVal);
-            } else {
-                options.plugins = [hotModuleVal];
-            }
+            options.plugins = [hotModuleVal];
```
Comment (rishabh3112) [inline packages/webpack-cli/lib/groups/AdvancedGroup.js:16]: This change disturbs immutability of  `options.plugins`.

File: packages/webpack-cli/lib/groups/AdvancedGroup.js [hunk 2]
```diff
@@ -13,21 +13,12 @@ class AdvancedGroup extends GroupHelper {
...
         }
         if (args.prefetch) {
             const { PrefetchPlugin } = require('webpack');
             const prefetchVal = new PrefetchPlugin(null, args.prefetch);
-            if (options && options.plugins) {
-                options.plugins.unshift(prefetchVal);
-            } else {
-                // Currently the Plugin function is not functional  -> https://github.com/webpack/webpack-cli/pull/1140#discussion_r376761359
-                // options.plugins = [prefetchVal];
-            }
```
Comment (snitin315) [inline packages/webpack-cli/lib/groups/AdvancedGroup.js:30]: @rishabh3112  I think we just need to uncomment this, I think he meant to comment `args.plugin` check here [https://github.com/webpack/webpack-cli/pull/1140#discussion_r376761359](https://github.com/webpack/webpack-cli/pull/1140#discussion_r376761359)

Comment (rishabh3112) [reply packages/webpack-cli/lib/groups/AdvancedGroup.js:30]: Yes, he meant `args.plugin` there.

File: packages/webpack-cli/lib/groups/AdvancedGroup.js [hunk 3]
```diff
@@ -22,11 +22,10 @@ class AdvancedGroup extends GroupHelper {
         if (args.prefetch) {
             const { PrefetchPlugin } = require('webpack');
             const prefetchVal = new PrefetchPlugin(null, args.prefetch);
-            if (options && options.plugins) {
```
Comment (rishabh3112) [inline packages/webpack-cli/lib/groups/AdvancedGroup.js:25]: Accessing `options.plugin` would raise error if `options` is undefined. It should be reverted imo.

Comment (snitin315) [reply packages/webpack-cli/lib/groups/AdvancedGroup.js:25]: on it

**Pull request merged**